big/2/public-tests/tests/mikolajkowe.rs

```diff
@@ -201,38 +199,44 @@ async fn concurrent_writes_are_serialized() {
     }
 
     // then
+    let mut receiving_set = tokio::task::JoinSet::new();
+    for _ in 0..2 {
+        let (mut receiver, _addr) = listener.accept().await.unwrap();
+        let hmac_system_key: [u8; 64] = config.hmac_system_key.clone().try_into().unwrap();
+        let hmac_client_key: [u8; 32] = config.hmac_client_key.clone().try_into().unwrap();
```

Comment:
I'd start listening before awaiting for response from system, but idk if that matters